Neos.Eel/Classes/Helper/ArrayHelper.php

```diff
@@ -342,15 +342,21 @@ public function pop(iterable $array): array
      *
      *     Array.push(array, e1, e2)
      *
-     * @param iterable $array
+     * @param iterable|scalar $array
      * @param mixed $element
      * @return array The array with the inserted elements
      */
-    public function push(iterable $array, $element): array
+    public function push($array, $element): array
```

Comment:
or should this better be target `master` with php 8 union types:

```suggestion
    public function push(iterable|string|int|float $array, $element): array
```

Reply:
I would not use union types. It can also be possible that you push Arrays, Booleans, Objects (Like node), etc.

Reply:
sorry i might have been missleading, this pr is not about the things ($element) to push in, but the $array where to push onto ...

so this pr is to loosen the `iterable $array` and make it more flexible to call `Array.push()` on not only traversable see: https://github.com/neos/flow-development-collection/issues/2710

currently this works:

`Array.push(['foo'], 'bar')`  -  `['foo', 'bar']`
`Array.push(['foo'], true)`  -  `['foo', true]`
`Array.push(['foo'], node)`  -  `['foo', node]`

i also want to allow this:

`Array.push('foo', 'bar')`  -  `['foo', 'bar']`

Reply:
We can check how it behaves with union types in Flow 8+ - we still don't properly support union types there when reflection is involved (which it isn't here IIRC).